### PR TITLE
Change from gTTS to tiktoks text to speech

### PR DESCRIPTION
**Pull request by The-Drobe**
Hello, I have added the ability to customize the voice. The end result is a much cleaner and nicer voice than gTTS text to speech.
### New Features
- Change the gender of the voice
- Multiple different voices to choose from
- Cleaner and less robotic voice

### CODE REVIEW

**Review by supervoko — APPROVED** [review]

**Comment by The-Drobe** [comment]
> Works like a charm! But are there more US Male voices? Thanks!

The voices that you can use are all listed [here](https://github.com/The-Drobe/RedditVideoMakerBot/blob/master/examples/ValidOptionsTTS.txt)

There is also this Twitter with voice samples but I have not tested if they work. [here](https://twitter.com/scanlime/status/1512598559769702406)

**Comment by noahcrum-dev** [comment]
Hey, sorry I am rather new to github, how do I download this and use it? thank you for creating!

**Comment by nickshanks347** [comment]
+1 for this, works great

File: video_creation/voices.py [hunk 1]
```diff
@@ -14,21 +16,30 @@ def save_text_to_mp3(reddit_obj):
     print_step("Saving Text to MP3 files...")
     length = 0
 
+    # load_dotenv
+    load_dotenv()
+    Voice = os.getenv('Voice')
+
+    # Set the voice depending on what was set in .env
+    if Voice == '':
```
Comment (MartinAlcalde) [inline video_creation/voices.py:24]: You should add `or Voice is None` because reading and empty string returns None

**Comment by The-Drobe** [comment]
> Could you add VoiceGender to main.py checks and setup.py?
> 
> Thanks!

Both have been done. When you mention VoiceGender I assume you mean the Voice variable.

**Comment by JasonLovesDoggo** [comment]
> * Change the gender of the voice

If you are going to flat-out copy my code you could at least give me credit for it.

**Comment by The-Drobe** [comment]
> > * Change the gender of the voice
> 
> If you are going to flat-out copy my code you could at least give me credit for it.

I referenced your code [here](https://github.com/The-Drobe/RedditVideoMakerBot/blob/master/video_creation/TTSwrapper.py#L3)

**Comment by JasonLovesDoggo** [comment]
I'm sorry I didn't see that comment but yeah that's honestly fine. Sorry again.

**Comment by JasonLovesDoggo** [comment]
> > > * Change the gender of the voice
> > 
> > If you are going to flat-out copy my code you could at least give me credit for it.
> 
> I referenced your code [here](https://github.com/The-Drobe/RedditVideoMakerBot/blob/master/video_creation/TTSwrapper.py#L3)   
> 
> Would you like to be referenced some where else as well?
> 
> It's also not exactly the same. If you wish to have more recognition I am a
> Happy to do that.

Oh also I updated that file to support unlimited text length so I would advise you to check that out

**Comment by The-Drobe** [comment]
> > > > * Change the gender of the voice
> > > 
> > > 
> > > If you are going to flat-out copy my code you could at least give me credit for it.
> > 
> > 
> > I referenced your code [here](https://github.com/The-Drobe/RedditVideoMakerBot/blob/master/video_creation/TTSwrapper.py#L3)
> > Would you like to be referenced some where else as well?
> > It's also not exactly the same. If you wish to have more recognition I am a
> > Happy to do that.
> 
> Oh also I updated that file to support unlimited text length so I would advise you to check that out

I did implement that myself on this fork. It is implemented very differently from how you do it tho.

**Comment by BlockArchitech** [comment]
Could you add VoiceGender to the checks in main.py and the setup.py assistant please?

**Comment by The-Drobe** [comment]
> Could you add VoiceGender to the checks in main.py and the setup.py assistant please?

In my code, the variable name `VoiceGender` has not been used since [this commit](https://github.com/The-Drobe/RedditVideoMakerBot/commit/9ba7f40d11c3279cccb4490c1d23b724d5293083) in all newer commits I use the `Voice` varible to better suite what what the varible is used for.

**Comment by DrMxrcy** [comment]
This branch has worked super well for me. Good work! Hopefully its implemented soon

**Comment by BlockArchitech** [comment]
>  I use the Voice varible to better suite what what the varible is used for.


Awesome! As long as it makes it to the setup assistant.

**Comment by The-Drobe** [comment]
> * Change the gender of the voice



> > I use the Voice varible to better suite what what the varible is used for.
> 
> Awesome! As long as it makes it to the setup assistant.

It was done in the following [commit](https://github.com/The-Drobe/RedditVideoMakerBot/commit/33f5f284ea177375ccc6daab1d4d88ea36104e4f#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R91)

**Comment by RobertSofronea** [comment]
Has anyone encountered this error: "mutagen.mp3.HeaderNotFoundError: can't sync to MPEG frame" if so how did you solve it?

**Review by BlockArchitech — CHANGES_REQUESTED** [review]
Please download the *latest* setup.py from the develop branch. This setup.py is invalid.

Your pull request should be `BASE: develop HEAD: develop`

**Comment by The-Drobe** [comment]
> Please download the *latest* setup.py from the develop branch. This setup.py is invalid.
> 
> Your pull request should be `BASE: develop HEAD: develop`

When I get the time I will. Rn in have to focus on assessments.

**Comment by DrMxrcy** [comment]
Not using the Setup.py, but everything else seems to work great

**Comment by Auto-bos** [comment]
Hey @The-Drobe I am a pretty new coder and decided to put your changes into the coding for this bot, but I get this error: "IndentationError: unindent does not match any outer indentation level"
Any help would be greatly appreciated!
Untitled

**Comment by JasonLovesDoggo** [comment]
I am happy to help
Can you take a wider screenshot of your code?
@Auto-bos

**Comment by Auto-bos** [comment]
> I am happy to help Can you take a wider screenshot of your code? @Auto-bos

I'm sorry I don't understand. That is all the text there is when I put "python main.py" into the command prompt of the folder

**Comment by Auto-bos** [comment]
> I am happy to help Can you take a wider screenshot of your code? @Auto-bos

Actually, I don't need the help. I managed to troubleshoot the problems myself!
being a very novice coder I am super proud of myself.
Thanks for offering the help however.

**Comment by The-Drobe** [comment]
I have changed the branch and updated the setup.py new pull request [here](https://github.com/elebumm/RedditVideoMakerBot/pull/467). I had to create a new pull request because you cant change the branch.

**Comment by CrackedBrick** [comment]
Any detailed guide? Sorry im new to coding, Github and stuff

**Comment by The-Drobe** [comment]
> Any detailed guide? Sorry im new to coding, Github and stuff

What do you need help with?

**Comment by CrackedBrick** [comment]
Changing the TTS voice, i just need a detailed guide how to change it

**Comment by The-Drobe** [comment]
> Changing the TTS voice, i just need a detailed guide how to change it

- You can change it in the .env file. Vaild options here  https://github.com/The-Drobe/RedditVideoMakerBot/blob/master/examples/ValidOptionsTTS.txt

**Pull request closed without merging**